nifi-extension-bundles/nifi-protobuf-bundle/nifi-protobuf-services/src/main/java/org/apache/nifi/services/protobuf/StandardProtobufReader.java

```diff
@@ -0,0 +1,306 @@
+    protected PropertyDescriptor buildSchemaTextProperty() {
+        return new PropertyDescriptor.Builder()
+            .fromPropertyDescriptor(SCHEMA_TEXT)
+            .required(true)
+            .clearValidators()
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .defaultValue("${proto.schema}")
+            .description("The text of a Proto 3 formatted Schema")
+            .build();
+    }
+
+    private RecordReader createProtobufRecordReader(final Map<String, String> variables, final InputStream in, final SchemaDefinition schemaDefinition) throws IOException {
+        final Schema schema = schemaCompiler.compileOrGetFromCache(schemaDefinition);
+        final ProtoSchemaParser schemaParser = new ProtoSchemaParser(schema);
+        final MessageName messageName = messageNameResolver.getMessageName(variables, schemaDefinition, in);
+        final RecordSchema recordSchema = schemaParser.createSchema(messageName.getFullyQualifiedName());
+        return new ProtobufRecordReader(schema, messageName.getFullyQualifiedName(), in, recordSchema);
+    }
+
+
+    private void setupMessageNameResolver(final ConfigurationContext context) {
+        final MessageNameResolverStrategyName messageNameResolverStrategyName = context.getProperty(MESSAGE_NAME_RESOLVER_STRATEGY).asAllowableValue(MessageNameResolverStrategyName.class);
+        messageNameResolver = switch (messageNameResolverStrategyName) {
+            case MESSAGE_NAME_PROPERTY -> new PropertyMessageNameResolver(context);
+            case MESSAGE_NAME_RESOLVER_SERVICE -> context.getProperty(MESSAGE_NAME_RESOLVER_CONTROLLER_SERVICE).asControllerService(MessageNameResolverService.class);
+        };
+    }
+
+    private SchemaDefinition createSchemaDefinitionFromText(final Map<String, String> variables) throws SchemaNotFoundException {
+        final String schemaTextString = schemaText.evaluateAttributeExpressions(variables).getValue();
+        validateSchemaText(schemaTextString);
+
+        final String sha256hex = sha256Hex(schemaTextString);
+        final SchemaIdentifier schemaIdentifier = SchemaIdentifier.builder()
+            .name(sha256hex + PROTO_EXTENSION)
+            .build();
+
+        return new StandardSchemaDefinition(schemaIdentifier, schemaTextString, SchemaDefinition.SchemaType.PROTOBUF);
+    }
+
+    private String sha256Hex(final String input) {
+        MessageDigest digest;
+        try {
+            digest = MessageDigest.getInstance("SHA-256");
+        } catch (NoSuchAlgorithmException e) {
+            throw new IllegalStateException(e);
+        }
+        byte[] hash = digest.digest(
+                input.getBytes(StandardCharsets.UTF_8));
+        final StringBuilder hexString = new StringBuilder();
+        for (byte b : hash) {
+            final String hex = Integer.toHexString(0xff & b);
+            if (hex.length() == 1)
+                hexString.append('0');
+            hexString.append(hex);
+        }
```

Comment:
This approach can be replaced [HexFormat](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/HexFormat.html)